src/gwt/src/org/rstudio/core/client/StringUtil.java

```diff
@@ -1165,10 +1166,20 @@ public static native int newlineCount(String str) /*-{
       return (str.match(/\n/g)||[]).length;
    }-*/;
    
+   /**
+    * Compare two strings, works if one or both strings are null.
+    * @param str1
+    * @param str2
+    * @return true if non-null strings are equal, or both are null
+    */
+   public static final boolean equals(String str1, String str2)
+   {
+      return Objects.equals(str1, str2);
```

Comment:
What do you think about using native JSNI here and directly calling `str1 == str2`? Just to avoid any potential overhead that `Object.equals()` might force upon us.